consensus/tests/block_queue.rs

```diff
@@ -221,6 +222,91 @@ async fn send_two_micro_blocks_out_of_order() {
     assert_eq!(blockchain1.get_block_at(2, true, None).unwrap(), block2);
 }
 
+#[tokio::test]
+async fn send_micro_blocks_out_of_order() {
+    let keypair =
+        KeyPair::from(SecretKey::deserialize_from_vec(&hex::decode(SECRET_KEY).unwrap()).unwrap());
+    let env1 = VolatileEnvironment::new(10).unwrap();
+    let env2 = VolatileEnvironment::new(10).unwrap();
+    let blockchain1 = Arc::new(Blockchain::new(env1, NetworkId::UnitAlbatross).unwrap());
+    let blockchain2 = Arc::new(Blockchain::new(env2, NetworkId::UnitAlbatross).unwrap());
+    let mut hub = MockHub::new();
+    let network = Arc::new(hub.new_network());
+    let mempool = Mempool::new(Arc::clone(&blockchain2), MempoolConfig::default());
+    let producer = BlockProducer::new(Arc::clone(&blockchain2), Arc::clone(&mempool), keypair);
+    let (request_component, mut mock_ptarc_rx, _mock_ptarc_tx) =
+        MockRequestComponent::<MockPeer>::new();
+    let (mut tx, rx) = mpsc::channel(32);
+
+    let mut block_queue = BlockQueue::with_block_stream(
+        Default::default(),
+        Arc::clone(&blockchain1),
+        network,
+        request_component,
+        rx.boxed(),
+    );
+
+    let mut rng = rand::thread_rng();
+    let mut ordered_blocks = Vec::new();
+
+    let mock_id = MockId::new(hub.new_address().into());
+
+    let n_Blocks = rng.gen_range(2, 15);
+
+    for n in 0..n_Blocks {
+        let block = Block::Micro(producer.next_micro_block(
+            blockchain2.time.now() + n * 1000,
+            0,
+            None,
+            vec![],
+            vec![0x42],
+        ));
+
+        // push it, so the producer actually produces a block
+        blockchain2.push(block.clone()).unwrap();
+
+        ordered_blocks.push(block);
+    }
+
+    let mut blocks = ordered_blocks.clone();
+
+    while blocks.len() > 1 {
+        let index = rng.gen_range(1, blocks.len());
+
+        tx.send((blocks.remove(index).clone(), mock_id.clone()))
+            .await
+            .unwrap();
+
+        // run the block_queue one iteration, i.e. until it processed one block
+        block_queue.poll_next_unpin(&mut Context::from_waker(noop_waker_ref()));
+    }
+
+    // All blocks should be buffered
+    assert_eq!(blockchain1.block_number(), 0);
+
+    //Obtain the buffered blocks
+    let buffered_blocks = block_queue.buffered_blocks().collect::<Vec<_>>();
+    assert_eq!(buffered_blocks.len() as u64, n_Blocks - 1);
+
+    // now send block1 to fill the gap
+    tx.send((blocks[0].clone(), mock_id)).await.unwrap();
+
+    for _i in 0..n_Blocks {
```

Comment:
Suggestion: Since you're not planning on using the `_i` variable, just use the underscore (`_`) to achieve the same effect